src/display/font_loader.js

```diff
@@ -365,9 +365,41 @@ var FontFaceObject = (function FontFaceObjectClosure() {
 
     getPathGenerator: function FontLoader_getPathGenerator(objs, character) {
       if (!(character in this.compiledGlyphs)) {
-        var js = objs.get(this.loadedName + '_path_' + character);
-        /*jshint -W054 */
-        this.compiledGlyphs[character] = new Function('c', 'size', js);
+        var cmds = objs.get(this.loadedName + '_path_' + character);
+        // Try to compile cmds into JS for MAXIMUM SPEED
+        var js = '';
+        var cmd, args;
+        try {
+          for (var i in cmds) {
+            cmd = cmds[i][0];
+            args = cmds[i].slice(1);
+            js += 'c.' + cmd + '(' + args.join(',') + ');\n';
+          }
+          /* jshint -W054 */
+          this.compiledGlyphs[character] = new Function('c', 'size', js);
+        } catch(e) {
+          // But fall back on using Function.prototype.apply() if we're blocked
+          // from using eval() for whatever reason (like CSP policies)
+          if (!(e instanceof EvalError)) {
+            throw e;
+          }
+          this.compiledGlyphs[character] = function(c, size) {
+            for (var i in cmds) {
+              cmd = cmds[i][0];
+              args = cmds[i].slice(1);
+
+              for (var j in args) {
+                if (args[j] === 'size') {
```

Comment:
I had it that way initially, I switched to this since it seems a bit more intuitive to me, especially if there are future changes to the font renderer. I'll change it back for performance.